### PR TITLE
Cleaned Dockerfile and locker version for openjdk7

### PR DESCRIPTION
**Pull request by amihaiemil**
PR for #84 .
Removed unnecessary settings and locked Java version. 
I set the same openjdk7 version as travis uses.

### CODE REVIEW

**Comment by amihaiemil** [comment]
@original-brownbear Can you maybe have a look, ask the PM to assign this to you? You're the docker guy as I noticed :D

**Comment by original-brownbear** [comment]
@amihaiemil I have no authority here (lol you got me scared for a a sec but I would've gotten some email about it at least I guess :D) but sure let me take a quick look :)

File: src/docker/Dockerfile [hunk 1]
```diff
@@ -56,39 +57,16 @@ RUN apt-get install -y software-properties-common python-software-properties
...
-RUN apt-add-repository ppa:brightbox/ruby-ng
-RUN apt-get update
-RUN apt-get install -y ruby2.2 ruby2.2-dev
-RUN gem update
-RUN gem install nokogiri
-RUN gem install bundler
-
 # Java
-RUN apt-get install -y default-jdk
+RUN apt-get install -y openjdk-7-jdk="${OPENJDK_VERSION}"
```
Comment (original-brownbear) [inline src/docker/Dockerfile]: @amihaiemil while you're at it, please make this `apt-get update && apt-get install -y ...`, see https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#/apt-get for an explanation ;)

File: src/docker/Dockerfile [hunk 2]
```diff
@@ -56,39 +57,16 @@ RUN apt-get install -y software-properties-common python-software-properties
...
-# SSH Daemon
-RUN apt-get install -y ssh && \
-  mkdir /var/run/sshd && \
-  chmod 0755 /var/run/sshd
-
-# Ruby
-RUN apt-get install -y libmagic-dev zlib1g-dev
-RUN apt-add-repository ppa:brightbox/ruby-ng
-RUN apt-get update
-RUN apt-get install -y ruby2.2 ruby2.2-dev
```
Comment (original-brownbear) [inline src/docker/Dockerfile]: @amihaiemil I don't think you can drop the ruby section, it's needed for the puzzle creation. You can make this a little nicer though (see below comment why this is important), by joining lines 67 and 68 into `RUN apt-get update && apt-get install -y ruby2.2 ruby2.2-dev`

**Comment by original-brownbear** [comment]
@amihaiemil check out a few small comments :)

**Comment by amihaiemil** [comment]
@original-brownbear done. Thanks a lot :D

**Comment by dmarkov** [comment]
@amihaiemil Thanks for the pull request, let me find a reviewer..

**Comment by dmarkov** [comment]
@mkordas please review this PR

**Comment by mkordas** [comment]
@amihaiemil I'm on it

File: src/docker/Dockerfile [hunk 3]
```diff
@@ -32,6 +32,7 @@ LABEL Description="This is the image for aether.jcabi.com" Vendor="jcabi.com" Ve
 WORKDIR /tmp
 
 ENV DEBIAN_FRONTEND=noninteractive
+ENV OPENJDK_VERSION 1.7.0_75
```
Comment (mkordas) [inline src/docker/Dockerfile]: @amihaiemil why not latest Java 7 here?

File: src/main/java/com/jcabi/aether/Aether.java [hunk 4]
```diff
@@ -94,10 +94,6 @@
  *  using jcabi-http, provide a maven configuration file with proxy settings
  *  and verify the expected HTTP requests have been sent to the mocked server.
  * @todo #69:30min Fix UnsupportedClassVersionError in CL build for Open JDK6.
- * @todo #82:30min Clean unnecessary docker setup (src/docler/Dockerfile),
- *  make the configuration minimal. Set versions to the used packages
- *  (e.g. lock java version instead of default jdk). After changes, the file
- *  needs to be uploaded to Dockerhub and be built there.
```
Comment (mkordas) [inline src/main/java/com/jcabi/aether/Aether.java]: @amihaiemil what about Dockerhub upload?

File: src/docker/Dockerfile [hunk 5]
```diff
@@ -56,39 +57,24 @@ RUN apt-get install -y software-properties-common python-software-properties
...
-RUN apt-get install -y libmagic-dev zlib1g-dev
+# Ruby; for puzzle creation
+RUN apt-get update && apt-get install -y \
+    libmagic-dev zlib1g-dev
 RUN apt-add-repository ppa:brightbox/ruby-ng
-RUN apt-get update
-RUN apt-get install -y ruby2.2 ruby2.2-dev
-RUN gem update
-RUN gem install nokogiri
-RUN gem install bundler
```
Comment (mkordas) [inline src/docker/Dockerfile]: @amihaiemil I think that `nokogiri` and `bundler` may also be required for puzzle creation

**Comment by mkordas** [comment]
@amihaiemil did you build project locally on this image? It is very important to check it

**Comment by mkordas** [comment]
@amihaiemil BTW, I think that decision was to move all Dockerfiles to https://github.com/yegor256/dockers where @yegor256 will run automatic image build for it

**Comment by mkordas** [comment]
@amihaiemil see my review above

File: src/main/java/com/jcabi/aether/Aether.java (hunk 4 above)
Comment (amihaiemil) [inline src/main/java/com/jcabi/aether/Aether.java]: @mkordas I was going to let Yegor know when merging.

File: src/docker/Dockerfile (hunk 3 above)
Comment (amihaiemil) [inline src/docker/Dockerfile]: @mkordas This is the same openjdk travis uses in builds, I figured it's better like this?

Comment (mkordas) [inline src/docker/Dockerfile]: @amihaiemil agreed, it's fine then

Comment (amihaiemil) [inline src/docker/Dockerfile]: @mkordas Actually, it's not - I tried building the image locally as u suggested and it cannot find the version ``1.7.0_75`` . I will look more into it today, to find a propper version. It sucks because Java versioning system was never quite clear to me... different names/numbers for the same thing etc etc. If you can suggest an openjdk7 version it would be cool :D

Comment (amihaiemil) [inline src/docker/Dockerfile]: @mkordas Should I simply say ``sudo apt-get install openjdk-7-jdk`` ? This what I find everywhere

**Comment by amihaiemil** [comment]
> BTW, I think that decision was to move all Dockerfiles to https://github.com/yegor256/dockers
@mkordas What do you mean? What should I do then?

**Comment by mkordas** [comment]
@amihaiemil I'll be here soon :)

**Comment by mkordas** [comment]
@amihaiemil starting :smile:

File: src/docker/Dockerfile (hunk 3 above)
Comment (amihaiemil) [inline src/docker/Dockerfile]: @mkordas Version of ``openjdk-7-jdk`` is ``7u111-2.6.7-0ubuntu0.14.04.3``, so I locked that one

**Comment by amihaiemil** [comment]
@mkordas Did the changes, the image builds fine locally with ``7u111-2.6.7-0ubuntu0.14.04.3`` openjdk7 version

File: src/main/java/com/jcabi/aether/Aether.java (hunk 4 above)
Comment (mkordas) [inline src/main/java/com/jcabi/aether/Aether.java]: @amihaiemil the thing is that @yegor256 always wanted PR to https://github.com/yegor256/dockers and then he was uploading... @original-brownbear do you remember why?

**Comment by mkordas** [comment]
@amihaiemil that's nice, thanks!

File: src/docker/Dockerfile (hunk 3 above)
Comment (mkordas) [inline src/docker/Dockerfile]: @amihaiemil this would be fine as well, it is very unlikely that new Java 7 version introduces breaking change :)

**Comment by amihaiemil** [comment]
@mkordas I say we proceed with this, let Yegor know and he will tell if I should make a PR for the other repo. But this could be merged and issue closed :D

File: src/docker/Dockerfile [hunk 6]
```diff
@@ -52,43 +52,31 @@ RUN apt-get update && apt-get install -y \
...
   wget --quiet http://ppa.launchpad.net/natecarlson/maven3/ubuntu/pool/main/m/maven3/maven3_3.2.1-0~ppa1_all.deb && \
   gdebi --non-interactive maven3_3.2.1-0~ppa1_all.deb && \
   ln -s /usr/share/maven3/bin/mvn /usr/bin/mvn && \
   rm -rf maven3_3.2.1-0~ppa1_all.deb
 
-# PhantomJS
-RUN apt-get install -y phantomjs
-
-# S3cmd for AWS S3 integration
-RUN apt-get install -y s3cmd
```
Comment (mkordas) [inline src/docker/Dockerfile]: @amihaiemil it's needed here: https://github.com/jcabi/jcabi-aether/blob/master/.rultor.yml#L38

File: src/docker/Dockerfile [hunk 7]
```diff
@@ -52,43 +52,31 @@ RUN apt-get update && apt-get install -y \
     automake autoconf
 
 # Git 2.0
-RUN apt-get install -y software-properties-common python-software-properties
+RUN apt-get update && apt-get install -y \
```
Comment (mkordas) [inline src/docker/Dockerfile]: @amihaiemil can't we just have somthing similar to https://github.com/yegor256/dockers/blob/master/xjm/Dockerfile? would be much shorter

File: src/docker/Dockerfile [hunk 8]
```diff
@@ -52,43 +52,31 @@ RUN apt-get update && apt-get install -y \
     automake autoconf
```
Comment (mkordas) [inline src/docker/Dockerfile]: @amihaiemil majority of these things are not needed, please compare with https://github.com/yegor256/dockers/blob/master/xjm/Dockerfile and check locally if anything is missing

File: src/docker/Dockerfile [hunk 9]
```diff
@@ -30,7 +30,7 @@ FROM ubuntu:14.04
 MAINTAINER Yegor Bugayenko <yegor@teamed.io>
 LABEL Description="This is the image for aether.jcabi.com" Vendor="jcabi.com" Version="1.0"
 WORKDIR /tmp
-
+ENV OPENJDK_VERSION 7u111-2.6.7-0ubuntu0.14.04.3
 ENV DEBIAN_FRONTEND=noninteractive
 
 # UTF-8 locale
```
Comment (mkordas) [inline src/docker/Dockerfile]: @amihaiemil I also don't think these locale stuff is necessary - @original-brownbear can you advise?

File: src/docker/Dockerfile [hunk 10]
```diff
@@ -52,43 +52,31 @@ RUN apt-get update && apt-get install -y \
...
 
-# Java
-RUN apt-get install -y default-jdk
-RUN apt-get install -y gdebi && \
+# Maven
+RUN apt-get update && apt-get install -y gdebi && \
   wget --quiet http://ppa.launchpad.net/natecarlson/maven3/ubuntu/pool/main/m/maven3/maven3_3.2.1-0~ppa1_all.deb && \
   gdebi --non-interactive maven3_3.2.1-0~ppa1_all.deb && \
   ln -s /usr/share/maven3/bin/mvn /usr/bin/mvn && \
   rm -rf maven3_3.2.1-0~ppa1_all.deb
```
Comment (mkordas) [inline src/docker/Dockerfile]: @amihaiemil can we go for the latest Maven as well?

File: src/docker/Dockerfile (hunk 7 above)
Comment (amihaiemil) [inline src/docker/Dockerfile]: @mkordas What exactly do you mean? :D I don't have so much experience with Docker and I don't know what this project needs exactly - I only removed what seemed clearly not needed to me. So just suggest changes if you know for sure they are ok :D

**Comment by mkordas** [comment]
@amihaiemil unfortunately not :( if we merge this PR no one will have interest with moving things forward... We need to complete it 100% or leave new puzzle

**Comment by mkordas** [comment]
@amihaiemil you can e.g. leave upload to Dockerhub and pinging Yegor in the next puzzle, that would be the optimal

**Comment by mkordas** [comment]
@amihaiemil see my review above

**Comment by amihaiemil** [comment]
>no one will have interest with moving things forward

@mkordas It's no bother for me to make a PR. Adding a puzzle would seem a bit rude - it's no effort...

File: src/docker/Dockerfile (hunk 7 above)
Comment (amihaiemil) [inline src/docker/Dockerfile]: @mkordas Im not sure what you want here

File: src/docker/Dockerfile (hunk 8 above)
Comment (amihaiemil) [inline src/docker/Dockerfile]: >and check locally if anything is missing

@mkordas check how? I only know to build the image and run it, how is that relevant to the project, and how it's built by rultor behing the scenes - I have no clue about that

**Comment by amihaiemil** [comment]
>You can e.g. leave upload to Dockerhub and pinging Yegor in the next puzzle, that would be the optimal

@mkordas Why cannot we close this, Yegor puts it up, everything is done, and then I simply make a PR with the Dockerfile for the other repo? No more puzzles just for this? I think that would be best. Or even Yegor can take the file and put it in which other repo he wants to

**Comment by amihaiemil** [comment]
@mkordas I added ``S3cmd`` and latest maven ``3.3.9``,  but I have a couple questions. Please check

**Comment by amihaiemil** [comment]
@mkordas ping

File: src/main/java/com/jcabi/aether/Aether.java (hunk 4 above)
Comment (original-brownbear) [inline src/main/java/com/jcabi/aether/Aether.java]: @mkordas actually putting the file to that repo only was the solution for commercial repos, because we didn't want to expose the whole repo publicly in that case.
For open source stuff, it should also be fine here, but i @yegor256 want's to keep things in one place that's his choice, works too :)

**Comment by mkordas** [comment]
@amihaiemil sorry, looks like this PR will require a lot of focus from my side, as the puzzle is really badly written. Will be here in a sec

**Comment by mkordas** [comment]
@amihaiemil OK, basically let's try to understand the goal of the task and when it can be closed: IMO it's only when the project **builds and deploys** successfully on Rultor with the new image. Everything until that state must be done here or puzzled out, otherwise no one will care and there will be image hanging around but nobody will use it

File: src/docker/Dockerfile (hunk 8 above)
Comment (mkordas) [inline src/docker/Dockerfile]: @amihaiemil after building the image, ssh to it, checkout sources and try out all commands used by rultor manually. I don't know any other method. If we don't do that, then the work is useless.

File: src/docker/Dockerfile (hunk 7 above)
Comment (mkordas) [inline src/docker/Dockerfile]: @amihaiemil I'm just suggesting to make this Docker file shorter. Instead of having many sections, you can have just three, e.g.
```dockerfile
RUN apt-get update && apt-get install -y software-properties-common \
  && apt-add-repository ppa:brightbox/ruby-ng

RUN apt-get update && apt-get install -y \
  openjdk-7-jdk \
  git \
  s3cmd \
  build-essential \
  libxml2-utils \
  libmagic-dev=1:5.14-2ubuntu3.3 \
  zlib1g-dev=1:1.2.8.dfsg-1ubuntu1 \
  ruby2.2=2.2.4-1bbox1~trusty1 \
  ruby2.2-dev=2.2.4-1bbox1~trusty1 \
  && rm -rf /var/lib/apt/lists/*

RUN gem update && gem install \
  nokogiri:1.6.7.2 \
  bundler:1.11.2 \
  pdd:0.15.4 \
  est:0.3.1
```
If you don't have Docker experience that's fine, you've already spend time here, you can puzzle out the rest

File: src/main/java/com/jcabi/aether/Aether.java (hunk 4 above)
Comment (mkordas) [inline src/main/java/com/jcabi/aether/Aether.java]: @original-brownbear now I remember,  thanks!

Comment (mkordas) [inline src/main/java/com/jcabi/aether/Aether.java]: @amihaiemil looks like we don't need to submit any PRs to https://github.com/yegor256/dockers, that's good

**Comment by original-brownbear** [comment]
@mkordas @amihaiemil just test it like this:

```bash
cd docker_dir # wherever you put the file
docker build . -t amihaimil/aether
cd jcabi-aether #project root
docker run -it -w $PWD -v "$PWD:PWD" --rm amihaimil/aether /bin/bash -lc 'mvn clean install site -Psite -Pqulice --errors --settings ../settings.xml;    mvn clean;    pdd --source=$(pwd) --verbose --file=/dev/null;  est --dir=est --file=/dev/null;'
```

done :)

**Comment by mkordas** [comment]
@amihaiemil 
> It's no bother for me to make a PR. Adding a puzzle would seem a bit rude - it's no effort...

But there's an effort. After image is build on Dockerhub successfully (and it may fail as well...) it needs to be applied for Rultor, and then Rultor must successfully merge and deploy the PR.

**Comment by amihaiemil** [comment]
@original-brownbear @mkordas thank you for clarifying. I'll look into it again this afternoon/evening

**Comment by mkordas** [comment]
@amihaiemil any success?

**Comment by amihaiemil** [comment]
@mkordas didnt manage to get to it today, sorry. Hopefully tomorrow

**Comment by mkordas** [comment]
@amihaiemil no problem, just let me know in case on any questions or if done :)

File: src/docker/Dockerfile (hunk 7 above)
Comment (amihaiemil) [inline src/docker/Dockerfile]: @mkordas I looked now into it... They already seem well grouped to me? Each tool/stack of tools with it's own, commented section and locked versions. 

[This](https://github.com/yegor256/dockers/blob/master/xjm/Dockerfile) example is much shorter because everything is messed up together

I think it's much readable the way it is now. You see everything nicely grouped rather than trying to read between the lines to figure out that eveything there does and if there is any relation between them.

Still want to regroup?

**Comment by amihaiemil** [comment]
@mkordas I did some regrouping and locked more versions, the same as in the example dockerfile... I will look into the locales stuff and test this evening. Please have a look.

**Comment by amihaiemil** [comment]
@mkordas I tested as bear said and it works fine without the locals stuff, so I took them out. I also took out git-core packages, leaving only git and it works fine. The versioning for ruby and rubydev doesn't work, it says it does not find them, so I left them as original.

**Comment by mkordas** [comment]
@amihaiemil sure, checking :)

File: src/docker/Dockerfile [hunk 11]
```diff
@@ -49,45 +46,26 @@ RUN apt-get update && apt-get install -y \
     bsdmainutils \
     libxml2-utils \
     build-essential \
-    automake autoconf
+    automake autoconf \
+    git \
```
Comment (mkordas) [inline src/docker/Dockerfile]: @amihaiemil what Git version is installed? can we fix to latest one as well?

**Comment by mkordas** [comment]
@amihaiemil 
> I did some regrouping and locked more versions, the same as in the example dockerfile...

It looks great to me, thanks!

**Comment by mkordas** [comment]
@amihaiemil what is the plan here? I don't see any new puzzle created, so how do we want to apply this image for Rultor builds?

File: src/docker/Dockerfile (hunk 7 above)
Comment (mkordas) [inline src/docker/Dockerfile]: @amihaiemil actually you are right, original version was grouped, but at least for me the shorter version is also pretty readable. There are groups, but not named with comments (it's visible what is being added anyway)

File: src/docker/Dockerfile [hunk 12]
```diff
@@ -49,45 +46,26 @@ RUN apt-get update && apt-get install -y \
...
+    s3cmd \
+    libmagic-dev=1:5.14-2ubuntu3.3 \
+    zlib1g-dev=1:1.2.8.dfsg-1ubuntu1 \
+    ruby2.2 \
+    ruby2.2-dev \
+    openjdk-7-jdk="${OPENJDK_VERSION}"
```
Comment (mkordas) [inline src/docker/Dockerfile]: @amihaiemil according to https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/, we should do ` && rm -rf /var/lib/apt/lists/*` after each `apt-get` block

File: src/docker/Dockerfile [hunk 13]
```diff
@@ -49,45 +46,26 @@ RUN apt-get update && apt-get install -y \
     bsdmainutils \
     libxml2-utils \
     build-essential \
-    automake autoconf
+    automake autoconf \
+    git \
+    s3cmd \
```
Comment (mkordas) [inline src/docker/Dockerfile]: @amihaiemil did you try locking it to `1.6.1`

**Comment by mkordas** [comment]
@amihaiemil see my comments above, only real minors + plan is left

**Comment by amihaiemil** [comment]
@mkordas I added the `` && rm -rf /var/lib/apt/lists/*`` after all apt-get bloks; I tried adding versions for git (2.9.3) and s3cmd (1.6.1), but none of them work, it says it cannot find them - so I left them as they were. 

I tested and it works fine. 

Now the next step is the same as [here](https://github.com/jcabi/jcabi-aether/pull/83) - @yegor256  has to build the image and push it on Dockerhub under his name. After that, no actions are necessary since it's already declared in ``.rultor.yml``

**Comment by amihaiemil** [comment]
@mkordas ping

**Comment by mkordas** [comment]
@amihaiemil 
> Now the next step is the same as here - @yegor256 has to build the image and push it on Dockerhub under his name. After that, no actions are necessary since it's already declared in .rultor.yml

Indeed, thanks - looks good

**Comment by mkordas** [comment]
@amihaiemil In order to specify git/s3cmd version I propose to run e.g. `apt-cache policy git`. The version to use should be displayed there. Also on http://snapshot.debian.org/package/git/1%3A2.9.3-1/ it's visible that `2.9.3-1` should be used instead of just `2.9.3`

**Comment by mkordas** [comment]
@amihaiemil anyway, these versions are minor, let's proceed then :)

**Comment by mkordas** [comment]
@rultor merge

**Comment by mkordas** [comment]
@yegor256 ping

**Comment by mkordas** [comment]
@yegor256 please take a look here as well

**Comment by mkordas** [comment]
@yegor256 ping

**Comment by mkordas** [comment]
@yegor256 can you upload image from the PR to the Dockerhub?

**Comment by mkordas** [comment]
@yegor256 ping

**Comment by mkordas** [comment]
@yegor256 your help with Dockerhub is needed here

**Comment by mkordas** [comment]
@yegor256 just a reminder :)

**Comment by mkordas** [comment]
@yegor256 ping

**Comment by amihaiemil** [comment]
@yegor256 ping

**Comment by mkordas** [comment]
@yegor256 please helo

**Comment by mkordas** [comment]
@yegor256 can you upload image to Dockerhub from here?

**Comment by amihaiemil** [comment]
@yegor256 Can u upload the new docker image to Dockerhub?

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by mkordas** [comment]
@yegor256 thanks, is the image updloaded to Dockerhub now?

**Comment by dmarkov** [comment]
@mkordas thanks, paid, **1 hour and 9 mins** to your account, payment ID is `57ea340ecd453c090e06cc3a`, task took 669 hours and 46 mins to complete... you're getting extra minutes here (c=54)... +69 added to your rating, current score is: [+2944](http://www.netbout.com/b/36080?open=rating)

**Comment by dmarkov** [comment]
@rultor deploy pls
